src/fontman/legacy_font.cpp

```diff
@@ -0,0 +1,443 @@
+            cc.tY = 0;
+            cc.tW = 15;
+            cc.tH = 17;
+            cc.width = 16;
+
+            if(c >= 48 && c <= 57)
+            {
+                cc.tX = (c - 48) * 16;
+                cc.texId = 0;
+            }
+            else if(c >= 65 && c <= 90)
+            {
+                cc.tX = (c - 55) * 16;
+                cc.texId = 0;
+            }
+            else if(c >= 97 && c <= 122)
+            {
+                cc.tX = (c - 61) * 16;
+                cc.texId = 0;
+            }
+            else if(c >= 33 && c <= 47)
+            {
+                cc.tX = (c - 33) * 16;
+                cc.texId = 1;
+            }
+            else if(c >= 58 && c <= 64)
+            {
+                cc.tX = (c - 58 + 15) * 16;
+                cc.texId = 1;
+            }
+            else if(c >= 91 && c <= 96)
+            {
+                cc.tX = (c - 91 + 22) * 16;
+                cc.texId = 1;
+            }
+            else if(c >= 123 && c <= 125)
+            {
+                cc.tX = (c - 123 + 28) * 16;
+                cc.texId = 1;
+            }
+
+            cc.valid = true;
+
+            m_charMap.insert({get_utf8_char(uchar), std::move(cc)});
+        }
+
+        m_isReady = true;
+        m_fontName = "font3";
+        break;
+
+    case 3: // Menu font
+        m_glyphWidth = 16;
+        m_glyphHeight = 18;
+
+        m_textures[0] = &GFX.Font2[2];
+
+        for(char c = 33; c <= 126; ++c)
+        {
+            char ucharH[5] = {c, 0, 0, 0, 0};
+            char ucharL[5] = {static_cast<char>(std::tolower(c)), 0, 0, 0, 0};
+
+            RasChar cc;
+            cc.texId = 0;
+            cc.tX = 2;
+            cc.tY = (c - 33) * 32;
+            cc.tW = 18;
+            cc.tH = 16;
+            cc.width = 18;
+
+            if(c == 'M')
+                cc.width += 2;
+
+            cc.valid = true;
+
+            m_charMap.insert({get_utf8_char(ucharH), cc});
+            m_charMap.insert({get_utf8_char(ucharL), std::move(cc)});
+        }
+
+        m_isReady = true;
+        m_fontName = "font1";
+        break;
+
+    case 4: // Message box font
+        m_glyphWidth = 18;
+        m_glyphHeight = 18;
+
+        m_textures[0] = &GFX.Font2[3];
+
+        for(char c = 33; c <= 126; ++c)
+        {
+            char ucharL[5] = {c, 0, 0, 0, 0};
+
+            RasChar cc;
+            cc.texId = 0;
+            cc.tX = 0;
+            cc.tY = (c - 33) * 16;
+            cc.tW = 18;
+            cc.tH = 16;
+            cc.width = 18;
+
+            if(c == 'M')
+                cc.width += 2;
+
+            cc.valid = true;
+
+            m_charMap.insert({get_utf8_char(ucharL), cc});
+        }
+
+        m_isReady = true;
+        m_fontName = "font2";
+        break;
+    }
+}
+
+PGE_Size LegacyFont::textSize(const char* text, size_t text_size,
+                              uint32_t max_line_lenght, bool cut,
+                              uint32_t /*fontSize*/)
+{
```

Comment:
How different is this from `RasterFont`'s implementation? I'd really encourage you to make `LegacyFont` inherit from `RasterFont` for code size and maintainability. (Same comment applies for later methods.)

Reply:
The difference is significant: the LegacyFont's is much simpler internally (in a case of the glyph alignment, size computation, etc.), and it's also different because it needs a totally different format of textures and other stuff.

Reply:
I see, then let's keep this part as-is. :+1: